src/jupyterlabcontroller/services/lab.py

```diff
@@ -94,31 +94,31 @@ async def info_event(self, username: str, message: str, pct: int) -> None:
             raise RuntimeError(
                 "% completion must be between 0 and 100 inclusive"
             )
-        ev_queue = self.event_manager.get(username)
+        event = Event(data=str(pct), type=EventType.PROGRESS)
+        self.event_manager.publish_event(username, event)
         umsg = f"{message} for {username}"
-        await ev_queue.asend(Event(data=str(pct), event=EventTypes.PROGRESS))
-        await ev_queue.asend(Event(data=umsg, event=EventTypes.INFO))
-        self.logger.info(f"Event: {umsg}: {pct}% ")
+        event = Event(data=umsg, type=EventType.INFO)
+        self.event_manager.publish_event(username, event)
+        self.logger.info(f"Event: {umsg}: {pct}%")
```

Comment:
Helpful during debugging (which is why I had it there) but I bet you're gonna want this at loglevel.DEBUG before long